spec/features/users_show_spec.rb

```diff
@@ -0,0 +1,67 @@
+require 'rails_helper'
+
+describe 'Users Show', type: :feature do
+  let :user do
+    User.create(name: 'Emi', photo: 'https://picsum.photos/100', bio: 'Football player')
+  end
+
+  let! :post1 do
+    Post.create(author_id: user.id, title: 'My life in football',
+                text: 'I am a player in the youth category for the Linces FBA team.')
+  end
+
+  let! :post2 do
+    Post.create(author_id: user.id, title: 'Receiver wing position',
+                text: 'The wing receiver position requires a lot of strength and speed.')
+  end
+
+  let! :post3 do
+    Post.create(author_id: user.id, title: 'Next season',
+                text: 'The next spring summer 2024 season will be the opportunity to improve my statistics.')
+  end
+
+  let! :comment1 do
+    Comment.create(user_id: user.id, post_id: post1.id, text: 'I like this post')
+  end
+
+  before { visit user_path(user) }
+
+  scenario "I can see the user's profile picture" do
+    expect(page).to have_css("img[src*='https://picsum.photos/100']")
+  end
+
+  scenario "I can see the user's username" do
+    expect(page).to have_content('Emi')
+  end
+
+  scenario 'I can see the number of posts the user has written' do
+    expect(page).to have_content('Number of posts: 3')
+  end
+
+  scenario "I can see the user's bio" do
+    expect(page).to have_content('Football player')
+  end
+
+  scenario "I can see the user's first 3 posts" do
+    expect(page).to have_content('My life in football')
+    expect(page).to have_content('Receiver wing position')
+    expect(page).to have_content('Next season')
+  end
+
+  scenario "I can see a button that lets me view all of a user's posts." do
+    expect(page).to have_link('See all posts')
+  end
+
+  scenario "When I click to see all posts, it redirects me to the user's post's index page." do
+    click_link 'See all posts'
+    expect(page).to have_content('My life in football')
+    expect(page).to have_content('Receiver wing position')
+    expect(page).to have_content('Next season')
+  end
+
+  scenario "When I click to see all posts, it redirects me to the user's post's index page" do
+    find('a', text: 'See all posts').click
+
+    expect(page).to have_current_path(user_posts_path(user.id))
+  end
```

Comment:
Your code looks good overall, but there are some changes that you need to make to meet the project requirements. Here are my suggestions:

- You have two scenarios that are testing the same thing: "When I click to see all posts, it redirects me to the user's post's index page." You should delete one of them to avoid duplication.
- You are missing a test that redirects to the post's show page when you click on a user's post. You should add this test to ensure that your feature works as expected.


I hope this helps you improve your code. Good luck with your project! 👍